### ISSUE TITLE
Assigning then unassigning ZRAM devices in Qubes Devices widget causes system lockup

### ISSUE THREAD

**Issue by Atrate** [QubesOS/qubes-issues#9255]
[How to file a helpful issue](https://www.qubes-os.org/doc/issue-tracking/)

### Qubes OS release

4.2.1

### Brief summary

AppVMs' zram devices show up in the Qubes Devices tray menu. Accidentally assigning them to an AppVM and then unassigning causes a lockup of most Qubes functionality (no new VMs can be launched, shut down, etc.).

### Steps to reproduce

1. Install `zram-generator` in a qube template (e.g. F39).

2. Configure `zram`, for example like this: 

```
[zram0]
zram-size = ram * 10
compression-algorithm = lz4
swap-priority = 200
fs-type = swap
```

(Note: I use `ram * 10` because zram-generator takes the amount of ram that it sees as the maximum at qube startup, this is usually around 800MB on my system, so it results in around 8GB of `zram` being created.) 

3. Shut the template down, boot a qube from that template (may need to reboot QubesOS first)

4. Click on the Qubes Devices tray menu, assign a `zram` device (called `()` in the menu) to a qube (I tried the same qube itself).

5. Unassign that `zram` device.

### Expected behavior

Zram devices shouldn't show up in the tray menu. If they do, it shouldn't break QubesOS to assign/unassign them.

### Actual behavior

Qubes Devices tray stops functioning, can't boot any new domains, can't shut any domains down, can't open Qube Manager, probably many other things I haven't tried. Shutting the OS down also takes a lot longer because of having to wait on unit timeouts.

**Comment by DemiMarie** [comment]
Wow, this is pretty bad.  I can understand that assigning them would break things, but this should not break the whole system!

**Comment by Atrate** [comment]
This is still an issue on `r4.3`

```
user@dom0 ~ % qvm-block
BACKEND:DEVID   DESCRIPTION                                                                USED BY
disp5764:zram0  Block_Storage: hosted by disp5764 zram0                                    
```

**Comment by alimirjamali** [comment]
RAM has become expensive. We would better solve this. It is possible to look for `zram` in device path [here](https://github.com/QubesOS/qubes-core-admin/blob/main/qubes/ext/block.py#L289-L311) and ignore it. I am not sure if there is any better solution.

**Comment by marmarek** [comment]
Better filter out on the udev side here: https://github.com/QubesOS/qubes-linux-utils/blob/main/udev/udev-qubes-block.rules

**Comment by DemiMarie** [comment]
Is filtering it out in the guest the right approach?  This seems like something in dom0 (probably qubesd) blocking synchronously on a qrexec call that never completes.

**Comment by marmarek** [comment]
Yes, it is the right approach. zram devices are not supposed to be attached to other VMs, there is no point in debugging symptoms of something that shouldn't happen in the first place.

**Comment by DemiMarie** [comment]
That makes sense.  Still, a buggy guest being able to hang the host this easily seems wrong.

**Comment by alimirjamali** [comment]
> Better filter out on the udev side here: https://github.com/QubesOS/qubes-linux-utils/blob/main/udev/udev-qubes-block.rules

PR to filter this out via udev rules is submitted. I tried `udevadm info --attribute-walk --path=/dev/zram0` and I do not see any other useful attribute to check for. So we could only depend on name.

> That makes sense. Still, a buggy guest being able to hang the host this easily seems wrong.

I believe this is correct from security perspective. Maybe reopening #9259 and following the crash issue separately there.
